runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/ProcessInstance/ProcessInstance.utils.ts

```diff
@@ -0,0 +1,21 @@
+import { IProcessInstance } from 'runbotics-common';
+
+import { ProcessInstanceState } from './ProcessInstance.state';
+
+export const updateProcessInstanceProps = (state: ProcessInstanceState, processInstance: IProcessInstance) => {
+    const { id, subProcesses, hasSubProcesses, isLoadingSubProcesses } = processInstance;
+    const pageContent = state.all.page?.content;
+    if (!pageContent) return;
+
+    state.all.page.content = pageContent
+        .map((instance: IProcessInstance) => 
+            instance.id !== id
+                ? instance 
+                : ({ 
+                    ...instance,
+                    subProcesses: subProcesses !== undefined ? subProcesses : instance.subProcesses, 
+                    hasSubProcesses: hasSubProcesses !== undefined ? hasSubProcesses : instance.hasSubProcesses,
+                    isLoadingSubProcesses: isLoadingSubProcesses !== undefined ? isLoadingSubProcesses : instance.isLoadingSubProcesses,
```

Comment:
id don't see the type for `isLoadingSubProcesses` in `ProcessInstanceState` interface. Weird if you didn't get any eslint error.
1. provide field for it in ProcessInstanceState
2. keep it outside of the process_instance object there should be another object next to it